static/src/js/components/GranuleResults/GranuleDownloadButton.jsx

```diff
@@ -104,37 +108,56 @@ export const GranuleDownloadButton = (props) => {
     }
   }
 
+  const stringifiedProjectParams = stringify({
+    ...params,
+    p,
+    pg
+  })
+
+  const downloadButtonProps = {
+    badge,
+    bootstrapVariant: 'primary',
+    className: 'granule-results-actions__download-all',
+    dataTestId: 'granule-results-actions__download-all-button',
+    disabled: granuleCount === 0 || initialLoading || disableDatabaseComponents === 'true',
+    icon: Download,
+    label: buttonText,
+    type: 'button',
+    variant: 'full'
+  }
+
+  if (!isLoggedIn) {
+    const projectPath = `${window.location.protocol}//${window.location.host}/projects${stringifiedProjectParams}`
+
+    return (
+      <PortalLinkContainer
```

Comment:
I am a little confused why we are setting the href directly, rather than using the `href` prop. The purpose of the `PortalLinkContainer` is to make sure that, if the user is currently in a portal, they remain in the portal after clicking the link so make sure you double check theres no regression there. 

If we do end up using the `onClick`, and not setting a `href` or a `to`, I think we could just set a `Button` component here.

Reply:
Ignore this. I think I am on the same page now. I just looked back at the component and remembered we added some logic in there that runs after an `onClick`

Reply:
I think I am still a little confused as to why were setting the location, rather than passing an `href`. Does that not accomplish what we want for some reason?

Reply:
@trevorlang I'm not sure if this is what you mean but, I tested the branch with `  <Button
        badge
        href={`${apiHost}/login?ee=${earthdataEnvironment}&state=${encodeURIComponent(projectPath)}`}
        bootstrapVariant="primary"
        className="granule-results-actions__download-all"
        dataTestId="granule-results-actions__download-all-button"
        disabled={granuleCount === 0 || initialLoading || disableDatabaseComponents === 'true'}
        icon={Download}
        label={buttonText}
        type="button"
        variant="full"
      >
        {buttonText}
      </Button>`
      locally and that was successful It does break the look of the button etc so if we were to change to match the one in `static/src/js/components/SecondaryToolbar/SecondaryToolbar.jsx` with a Button there'd be additional work there

Reply:
I accidentally thought that we were calling `onChangePath` for every `onClick` in a `PortalLinkContainer` but that appears to not be the case. If replacing with a `Button` works, it seems like it would be the preferred method. I'm not sure that the `PortalLinkContainer` is providing any additional behavior in this case. I'm surprised there are styling differences as the `PortalLinkContainer` just ultimately renders a `Button`. Not trying to overcomplicate the PR but I don't think we should be using more complicated components than we need to if possible.

Reply:
I believe this removes an intermediate page from the redirect, and makes the user experience cleaner.